index.ts

```diff
@@ -1,35 +1,65 @@
 export class OatyArray {
-  private _original: object[]
-  private _keys: string[]
-  private _transposed: object = {}
+  private keys: string[] | undefined
+  private original: object[] = []
+  private transposed: { [key: string]: any } = {}
+
+  constructor(options?: { data?: object[], keys?: string[] }) {

```

Comment:
I'm not too keen on `OatyArray` only taking an options object as a parameter. Would prefer the initial array and an `options` object, e.g.

```ts
constructor(input?: array, options?: { keys?: string[] }) {
```

so you can init a new instance as, for example:
```ts
const empty = new OatyArray()
const emptyWithKeys = new OatyArray([], { keys: ['a', 'b'] })
const arrayNoKeys = new OatyArray([...])
```

Reply:
As it stands you can omit or redorder the args, like this:
`new OatyArray()`
`new OatyArray({})`
`new OatyArray({ keys: ['a', 'b'] })`
`new OatyArray({ data: [...] })`
`new OatyArray({ data: [...], keys: ['a', 'b'] })`
`new OatyArray({ keys: ['a', 'b'], data: [...] })`

This is more future proof and keeps it customisable for when we add more options.

Reply:
I understand that, but think the `options` parameter should be for options, rather than options _and_ data.

Reply:
Ohhh I'm with you now, agreed.

Reply:
Resolved